File: app/src/main/java/com/alterneo/alterneo_app/responses/CompaniesResponse.java
package com.alterneo.alterneo_app.responses;

import java.util.List;

public class CompaniesResponse {

	String total_returned;
	List<CompanyResponse> data;

	public CompaniesResponse(String total_returned, List<CompanyResponse> data) {
		this.total_returned = total_returned;
		this.data = data;
	}

	public String getTotal_returned() {
		return total_returned;
	}

	public void setTotal_returned(String total_returned) {
		this.total_returned = total_returned;
	}

	public List<CompanyResponse> getData() {
		return data;
	}

	public void setData(List<CompanyResponse> data) {
		this.data = data;
	}
}

File: app/src/main/java/com/alterneo/alterneo_app/responses/CompaniesResponse.kt
package com.alterneo.alterneo_app.responses

import com.alterneo.alterneo_app.models.Company

class CompaniesResponse(
    var total_returned: Number,
    var data: List<CompanyResponse>
) : ArrayResponse<CompanyResponse>(total_returned, data) {
    fun toCompanies(): List<Company> {
        val toReturn = ArrayList<Company>()
        data.forEach { toReturn.add(it.toCompany()) }
        return toReturn
    }
}